src/Microsoft.CorrelationVector/CorrelationVector.cs

```diff
@@ -1,397 +1,122 @@
-        }
-
-        private CorrelationVector(string baseVector, int extension, CorrelationVectorVersion version, bool immutable)
-        {
-            this.BaseVector = baseVector;
-            this.extension = extension;
-            this.Version = version;
-            this.immutable = immutable || CorrelationVector.IsOversized(baseVector, extension, version);
-        }
-
-        private static string GetUniqueValue(CorrelationVectorVersion version)
-        {
-            if (CorrelationVectorVersion.V1 == version)
-            {
-                byte[] bytes = Guid.NewGuid().ToByteArray();
-                return Convert.ToBase64String(bytes, 0, 12);
-            }
-            else if (CorrelationVectorVersion.V2 == version)
-            {
-                return Guid.NewGuid().GetBaseFromGuid();
-            }
-            else
-            {
-                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, "Unsupported correlation vector version: {0}", version));
-            }
-        }
-
-        private static CorrelationVectorVersion InferVersion(string correlationVector, bool reportErrors)
-        {
-            int index = correlationVector == null ? -1 : correlationVector.IndexOf('.');
-
-            if (CorrelationVector.BaseLength == index)
-            {
-                return CorrelationVectorVersion.V1;
-            }
-            else if (CorrelationVector.BaseLengthV2 == index)
-            {
-                return CorrelationVectorVersion.V2;
-            }
-            else
-            {
-                //By default not reporting error, just return V1
-                return CorrelationVectorVersion.V1;
-            }
-        }
-
-        private static bool IsImmutable(string correlationVector)
+                //By default not reporting error, just return V1
+                return CorrelationVectorVersion.V1;
+            }
+        }
+
+        /// <summary>
+        /// Converts a string representation of a Correlation Vector into this class.
+        /// </summary>
+        /// <param name="correlationVector">String representation</param>
+        /// <returns>The Correlation Vector based on its version.</returns>
+        public static CorrelationVector Parse(string correlationVector)
+        {
+            CorrelationVectorVersion version = CorrelationVector.InferVersion(correlationVector);
+            return RunStaticMethod(correlationVector, version, CorrelationVectorV1.Parse, CorrelationVectorV2.Parse);
+
+        }
+
+        public static CorrelationVector Extend(string correlationVector)
+        {
+            CorrelationVectorVersion version = CorrelationVector.InferVersion(correlationVector);
+            return RunStaticMethod(correlationVector, version, CorrelationVectorV1.Extend, CorrelationVectorV2.Extend);
+
+        }
+        public static CorrelationVector Spin(string correlationVector)

```

Comment:
Let's remove this static spin method and set a default for the SpinParameters in the other method

Reply:
This is how the method is set up in the original CorrelationVector class definition.

Reply:
In that case, create the default parameters in this static method and call the other static method. That would be a cleaner approach